test/mpi/configure.ac

```diff
@@ -1438,6 +1438,10 @@ fi
 if test "$pac_cv_have_mpi_integer16" = yes ; then 
     AC_DEFINE(HAVE_MPI_INTEGER16,1,[Define if MPI_INTEGER16 is available])
 fi
+if test "$FROM_MPICH" = yes ; then
```

Comment:
Why bother with this instead of relying on the test below? Can this patch be dropped?

Reply:
Unfortunately, you cannot ... `#include <mpi.h>` always fail from MPICH and regardless the appropriate flags were passed or not. I fixed my last commit (and rebased everything), and hopefully, we should be all set (and the last commit will be kept)

Reply:
If you run `configure` from the MPICH top directory, here is what you currently find in `test/mpi/config.log`

```
configure:12304: checking whether MPI_2DOUBLE_COMPLEX is available
configure:12325: gcc -std=gnu99 -c    conftest.c >&5
conftest.c:52:25: fatal error: mpi.h: No such file or directory
         #include "mpi.h"
                         ^
compilation terminated.
configure:12325: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "mpich-testsuite"
| #define PACKAGE_TARNAME "mpich-testsuite"
| #define PACKAGE_VERSION "3.3b3"
| #define PACKAGE_STRING "mpich-testsuite 3.3b3"
| #define PACKAGE_BUGREPORT "discuss@mpich.org"
| #define PACKAGE_URL "http://www.mpich.org/"
| #define PACKAGE "mpich-testsuite"
| #define VERSION "3.3b3"
| #define HAVE_MPI_SPAWN 1
| #define HAVE_MPI_WIN_CREATE 1
| #define HAVE_LONG_DOUBLE 1
| #define HAVE_LONG_LONG 1
| #define restrict __restrict
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_STDARG_H 1
| #define HAVE_SYS_TIME_H 1
| #define HAVE_SYS_RESOURCE_H 1
| #define HAVE__BOOL 1
| #define HAVE_FLOAT__COMPLEX 1
| #define HAVE_DOUBLE__COMPLEX 1
| #define HAVE_LONG_DOUBLE__COMPLEX 1
| #define SIZEOF_VOID_P 8
| #define USE_LONG_DOUBLE_COMPLEX 1
| #define HAVE_PTHREAD_H 1
| #define HAVE_PTHREAD_CREATE 1
| #define HAVE_PTHREAD_YIELD 1
| #define HAVE_PTHREAD_BARRIER_INIT 1
| #define HAVE_PTHREAD_BARRIER_WAIT 1
| #define HAVE_H_ADDR_LIST 1
| #define HAVE_GETRUSAGE 1
| #define HAVE_FORTRAN_BINDING 1
| #define STDCALL
| #define F77_NAME_LOWER_USCORE 1
| #define HAVE_IOSTREAM 1
| #define HAVE_NAMESPACES /**/
| #define HAVE_NAMESPACE_STD /**/
| #define HAVE_MPI_IO 1
| #define MPIO_USES_MPI_REQUEST /**/
| #define HAVE_MPI_INTEGER16 1
| /* end confdefs.h.  */
|
|         #include "mpi.h"
| int
| main ()
| {
|
|             MPI_Datatype t = MPI_2DOUBLE_COMPLEX;
|
|   ;
|   return 0;
| }
|
configure:12337: result: no
```

the test fail (this is fine since `MPI_2DOUBLE_COMPLEX` is not available out of the box) but for a bad reason (`fatal error: mpi.h: No such file or directory`), and it will always fail when invoked from MPICH. That is why, from MPICH, we cannot use `mpi.h` but we should use the macro directly.

makes sense ?

Reply:
Yes, now I understand the problem. I still don't like the solution, since the underlying problem is not solved. I still prefer to drop this patch and work on a solution to properly accessing `mpi.h`.